### ISSUE TITLE
Routing V2 sub-route outlet doesn't show embedded in parent route

### ISSUE THREAD

**Issue by MajorTom327** [remix-run/remix#5170]
### What version of Remix are you using?

1.11.1

### Steps to Reproduce

I'ved make a simple repo for the reproduction of this problem:
https://github.com/MajorTom327/remix-route-v2-tests

The routes presents are:
```
profile/index.tsx
profile.general/index.tsx
profile.security/index.tsx
_index.tsx
```

The `profile/index.tsx` is a simple route that have 2 NavLink to the route general and security.
`profile.general/index.tsx` and `profile.security` are simple subroute with different text to show.

The objective is to make a interface with pane or tab to navigate through sub-sections of the profile.

Click on any link in the profile section show the correct page but the links in the parent route disappear.

### Expected Behavior

The parent route should still be visible with the nav-links inside them. The child route should be in the Outlet

### Actual Behavior

Navigate through the sub-route work, but the parent layout is not show.

**Comment by xHomu** [comment]
It appears folders exhibit v1 behavior, even with v2_routeConvention flag.

**Comment by MajorTom327** [comment]
That can explain my secondary problem:

```
profile/index.tsx
profile.[_general]/index.tsx
profile.[_security]/index.tsx
```

Throwing an error on build like the escaping special chars doesn't work either.
<img width="936" alt="image" src="https://user-images.githubusercontent.com/16199396/214027681-126444d8-d64b-4ef0-850a-6d5f34c4ca88.png">

**Comment by MajorTom327** [comment]
A way to bypass the original problem:
Keep your parent route as a file, but you can use children route with the directory version

Like this:
```
profile.tsx
profile.general/index.tsx
profile.security/index.tsx
_index.tsx
```

That work fine. (But still not ok for `profile` route as a directory)

**Comment by MichaelDeBoey** [comment]
Closed by #5228
